backends/tofino/bf-p4c/midend/desugar_varbit_extract.h

```diff
@@ -123,7 +123,7 @@ class AnnotateVarbitExtractStates : public Transform {
             if (!method) continue;
 
             if (method->member == "extract" && call->arguments->size() == 2) {
-                state->addOrReplaceAnnotation("dontmerge"_cs, {});
+                state->addOrReplaceAnnotation(new IR::Annotation("dontmerge"_cs, {}));
```

Comment:
This seems like a pointless change

Reply:
Without this change, a later visitor traversal will cause a compiler bug. There might be something wrong with the construction of the annotation here.

Reply:
The problem is that {} is not a good `const IR::Expression*` argument. Not sure why this compiles, maybe decays to a null pointer?

Reply:
Looks like we're missing the overload of `addOrReplaceAnnotation` that takes a `std::initializer_list`.  It is weird that it doesn't give an error -- `{}` should not be a valid pointer.

Reply:
It looks like the Annotation interface is a little dirty, ideally we should find these issues at compile time. 

I could try to delete the overloaded nullpointer signature or see whether I can enforce references instead.